cmd/client/client.go

```diff
@@ -13,137 +13,128 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-
-// Package client creates a tunnel client to proxy incoming connections
+//
+// This binary creates a tunnel client to proxy incoming connections
 // over a grpc transport.
-package client
+// Exmaples to use this binary with ssh's ProxyCommand option:
+// TLS:
+//		ssh -o ProxyCommand="client
+//		--tunnel_server_address=localhost:$PORT \
+//		--cert_file=$CERT_FILE \
+//		--dial_target=target1 \
+//		--dial_target_type=SSH" $USER@localhost
+// mTLS:
+//		ssh -o ProxyCommand="client
+//		--tunnel_server_address=localhost:$PORT \
+//		--cert_file=$CERT_FILE \
+//		--key_file=$KEY_FILE \
+//		--ca_file=$CA_FILE \
+//		--dial_target=target1 \
+//		--dial_target_type=SSH" $USER@localhost
+package main
 
 import (
 	"context"
+	"flag"
 	"fmt"
 	"io"
 	"log"
-	"net"
+	"os"
+	"sync"
+	"time"
 
+	"github.com/cenkalti/backoff/v4"
 	"github.com/openconfig/grpctunnel/bidi"
 	"github.com/openconfig/grpctunnel/tunnel"
-
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
 
 	tpb "github.com/openconfig/grpctunnel/proto/tunnel"
 )
 
-// Config defines the parameters to run a tunnel client.
-type Config struct {
-	TunnelAddress, DialAddress, ListenAddress, CertFile, Target, TargetType string
-}
+var (
+	tunnelAddress  = flag.String("tunnel_server_address", "", "The address of the tunnel")
+	dialTarget     = flag.String("dial_target", "", "The client uses target to register at the server.")
+	dialTargetType = flag.String("dial_target_type", "", "The type of target protocol, e.g. GNMI or SSH.")
+	certFile       = flag.String("cert_file", "", "The certificate file location. If both cert_file and key_file are provided, mTLS will be used.")
+	keyFile        = flag.String("key_file", "", "The private key file location. If both cert_file and key_file are provided, mTLS will be used.")
+	caFile         = flag.String("ca_file", "", "The CA file location (for mTLS)")
```

Comment:
The help message here should be update to not include "(for mTLS)"

Reply:
Fixed.